commit 7d8fa9789ec3a8ba09a26726b8101e46a3e60d3c
Author: bertin <macportsuser@macports.org>
Date:   Sun May 25 22:21:48 2025 +0200

    /opt/local/site-ports/sysutils/MacPorts/files/patch-platform-command.diff

diff --git a/src/port1.0/portutil.tcl b/src/port1.0/portutil.tcl
index b48ac479b..f9450caba 100644
--- a/src/port1.0/portutil.tcl
+++ b/src/port1.0/portutil.tcl
@@ -732,30 +732,38 @@ proc variant_exists {name} {
 # Portfile level procedure to provide support for declaring platform-specifics
 # Basically, just a fancy 'if', so that Portfiles' platform declarations can
 # be more readable, and support arch and version specifics
-proc platform {args} {
+proc platform {os args} {
     global os.platform os.subplatform os.arch os.major
 
-    set len [llength $args]
-    if {$len < 2} {
+    if {[llength $args] < 1} {
         return -code error "Malformed platform specification"
     }
-    set code [lindex $args end]
-    set os [lindex $args 0]
-    set args [lrange $args 1 end-1]
+    set len 1
+    if {[lindex $args end-1] eq "else"} {
+        set code [lindex $args end-2]
+        set altcode [lindex $args end]
+        set consumed 3
+    } else {
+        set code [lindex $args end]
+        set altcode ""
+        set consumed 1
+    }
 
     set release_re {(^[0-9]+$)}
     set arch_re {([a-zA-Z0-9]*)}
-    foreach arg $args {
+    foreach arg [lrange $args 0 end-$consumed]  {
         if {[regexp $release_re $arg match result]} {
             set release $result
+            set len [expr $len + 1]
         } elseif {[regexp $arch_re $arg match result]} {
             set arch $result
+            set len [expr $len + 1]
         }
     }
 
     set match 0
     # 'os' could be a platform or an arch when it's alone
-    if {$len == 2 && ($os eq ${os.platform} || $os eq ${os.subplatform} || $os eq ${os.arch})} {
+    if {$len == 1 && ($os eq ${os.platform} || $os eq ${os.subplatform} || $os eq ${os.arch})} {
         set match 1
     } elseif {($os eq ${os.platform} || $os eq ${os.subplatform})
               && (![info exists release] || ${os.major} == $release)
@@ -766,6 +774,8 @@ proc platform {args} {
     # Execute the code if this platform matches the platform we're on
     if {$match} {
         uplevel 1 $code
+    } elseif {${altcode} ne ""} {
+        uplevel 1 $altcode
     }
 }
 
